### DISCUSSION TITLE
Use arrays for common conditions.

### DISCUSSION THREAD

**Discussion by h2zero** [theengs/decoder#167] [Ideas]
There are many cases when the decoders check for certain data at the same location of a parameter with different values.
For instance: https://github.com/theengs/decoder/blob/b752cfcf7c078867dc408d71b53a669187b6c46f/src/devices/CGD1_json.h#L10

This could be possibly be simplified to use a json array. Using the example above:
`"condition":["servicedata", "index", 0, "080caf", "|", "servicedata", "index", 0, "080c09", "|", "servicedata", "index", 0, "080cd0"]`

Could be potentially be converted to:
`"condition":["servicedata", "index", 0, ["080caf", "080c09", "080cd0"]]`

This could be added as an extension to #99

**Comment by DigiH** [comment]
We just have to be careful to document it clearly, as on its own with a few chained ORs it's fine, but when a pseudo-nested AND gets added, which might even have additional ORs included it can easily get a bit confusing again.

**Reply by h2zero** [reply]
Yes, it may be a limited use item for cases such as shown above. Just a thought anyway.
